receiver/azureeventhubreceiver/azureresourcetraces_unmarshaler.go

```diff
@@ -0,0 +1,183 @@
+package azureeventhubreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver"
+
+import (
+	"bytes"
+	"fmt"
+
+	"encoding/hex"
+	"net/url"
+
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.13.0"
+	"go.uber.org/zap"
+)
+
+type azureResourceTracesUnmarshaler struct {
+	buildInfo component.BuildInfo
+	logger    *zap.Logger
+}
+
+type azureTracesRecords struct {

```

Comment:
Starting to unmarshal at the `Records` level into a `[]azureTracesUnmarshaler` instead of a enveloping a single-field.

Reply:
Those records are processed here.

https://github.com/deepan10/opentelemetry-collector-contrib/blob/0206495a55ca3203a93035bf016717f4cbb8876c/receiver/azureeventhubreceiver/azureresourcetraces_unmarshaler.go#L90